src/game.ts

```diff
@@ -1,13 +1,159 @@
-class Game {
-  boardSize: number;
+import { Flow } from "./flow.js";
+import { Color, Direction, Point, Share } from "./models/models.js";
+import {
+  directions,
+  getValidFlow,
+  isCorner,
+  isfinishFlow,
+  isInvalidNewDot,
+  isOnAnotherDot,
+  isValidNewLine,
+  noLineWrap,
+  pointsAreEqual,
+  touchingOtherFlowDot,
+  touchingOtherFlowLine,
+} from "./utils.js";
+
+export class Game {
+  boardSize: number = 8;
   flows: Flow[] = [];
   showSolution: boolean = false;
   currentFlow?: Flow;
+  timeStart: Date = new Date();
+  movesCounter: number = 0;
+
+  public reset(): void {
+    this.movesCounter = 0;
+    this.flows = [];
+    this.currentFlow = undefined;
+    this.showSolution = false;
+  }
+
+  public createLevel(): void {
+    this.reset();
+    this.initializeLevel();
+
+    const iterationsMax = 10000;
+    for (let i = 0; i < iterationsMax; ) {
+      let flow: Flow = getValidFlow(this.flows, this.boardSize);
+      let currentDot: Point =
+        Math.floor(Math.random() * 2) == 0 ? flow.start : flow.end;
+
+      let direction: Direction = directions.at(Math.floor(Math.random() * 4))!;
```

Comment:
`Property 'at' does not exist on type 'Direction[]'. Do you need to change your target library? Try changing the 'lib' compiler option to 'es2022' or later.ts(2550)`

Reply:
fait